Fix doctests and add keyword arguments in factors submodule

Vanshitaaa20 opened the pull request:
## Your checklist for this pull request
Please review the [guidelines for contributing](CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [x] Make sure you are making a pull request against the **dev branch** (left side). Also you should start *your branch* off *our dev*.
- [x] Check the commit's or even all commits' message styles matches our requested structure.

### Issue number(s) that this pull request fixes
-  #757  
- #581 


### List of changes to the codebase in this pull request

- Corrected doctest examples and outputs in:
  - pgmpy/factors/base.py
  - pgmpy/factors/FactorSet.py
  - pgmpy/factors/continuous/LinearGaussianCPD.py
  - pgmpy/factors/discrete/CPD.py
  - pgmpy/factors/discrete/DiscreteFactor.py
  - pgmpy/factors/discrete/JointProbabilityDistribution.py
  - pgmpy/factors/discrete/NoisyOR.py
  - pgmpy/factors/hybrid/FunctionalCPD.py
  
- Replaced positional arguments with keyword arguments for clarity and readability in doctests.
- Added inline comments in doctests to explain where sorting is applied before assertions, ensuring consistent output.
- Addressed non-deterministic outputs caused by:
  - Unordered data structures like sets.
  - Memory addresses in object representations.
  by applying doctest: +ELLIPSIS or sorting techniques.

## ⚠ Additional Notes

Skipped memory addresses and adjusted set order outputs to prevent test inconsistencies across environments.

Vanshitaaa20 commented:
@ankurankan I pushed my changes after running pre-commit locally and everything seemed fine, but the GitHub checks are still failing. I’m not exactly sure what’s going wrong

ankurankan commented:
@Vanshitaaa20 Did you run black on the whole repository? If you look into the failing logs, it tells you the file that needs to be formatted: 
```
Run black --diff .
Skipping .ipynb files as Jupyter dependencies are not installed.
You can fix this by running ``pip install "black[jupyter]"``
error: cannot format /home/runner/work/pgmpy/pgmpy/pgmpy/estimators/__init__.py: Cannot parse for target version Python 3.[1](https://github.com/pgmpy/pgmpy/actions/runs/15498613388/job/43641172615?pr=2181#step:9:1)3: 20:3: fix/factors-doctests

Oh no! 💥 💔 💥
162 files would be left unchanged, 1 file would fail to reformat.
Error: Process completed with exit code [12](https://github.com/pgmpy/pgmpy/actions/runs/15498613388/job/43641172615?pr=2181#step:9:13)3.
```
Because you have also ran black on a lot more files than required, you will see a circular import error after you fix the above error. You will need to revert back `pgmpy/estimators/__init__.py` file to the original version.

Vanshitaaa20 committed:
Fix doctests for factor submodule

Vanshitaaa20 commented:
> @Vanshitaaa20 Did you run black on the whole repository? If you look into the failing logs, it tells you the file that needs to be formatted:
> 
> ```
> Run black --diff .
> Skipping .ipynb files as Jupyter dependencies are not installed.
> You can fix this by running ``pip install "black[jupyter]"``
> error: cannot format /home/runner/work/pgmpy/pgmpy/pgmpy/estimators/__init__.py: Cannot parse for target version Python 3.[1](https://github.com/pgmpy/pgmpy/actions/runs/15498613388/job/43641172615?pr=2181#step:9:1)3: 20:3: fix/factors-doctests
> 
> Oh no! 💥 💔 💥
> 162 files would be left unchanged, 1 file would fail to reformat.
> Error: Process completed with exit code [12](https://github.com/pgmpy/pgmpy/actions/runs/15498613388/job/43641172615?pr=2181#step:9:13)3.
> ```
> 
> Because you have also ran black on a lot more files than required, you will see a circular import error after you fix the above error. You will need to revert back `pgmpy/estimators/__init__.py` file to the original version.

I accidentally ran black on more files ... I fixed the formatting only for the files I worked on and removed unused imports and variables that were causing issues while running black.
I’ll be careful about running black selectively on just the changed files from next time. Thanks for pointing it out!

ankurankan commented:
@Vanshitaaa20 I also see that you have done a force push. Were you getting merge conflicts? This has reverted some changes in `FunctionalCPD.py`: https://github.com/pgmpy/pgmpy/pull/2181/files#diff-510efa4f418a01bfa5fe3b6f37b7b27f69f442194cfcb29b3e544b35523c311bL45. And because of that tests are failing. You will have to revert these changes (simplest way would be to copy-paste the current `FunctionalCPD.py` from github and make your doctest changes again in this file). In general, you should always avoid doing a force push unless you exactly know what it is going to do.

Vanshitaaa20 commented:
> @Vanshitaaa20 I also see that you have done a force push. Were you getting merge conflicts? This has reverted some changes in `FunctionalCPD.py`: https://github.com/pgmpy/pgmpy/pull/2181/files#diff-510efa4f418a01bfa5fe3b6f37b7b27f69f442194cfcb29b3e544b35523c311bL45. And because of that tests are failing. You will have to revert these changes (simplest way would be to copy-paste the current `FunctionalCPD.py` from github and make your doctest changes again in this file). In general, you should always avoid doing a force push unless you exactly know what it is going to do.

Yes, I was running into some merge conflicts earlier and ended up doing a force push while trying to resolve them.
 Noted about avoiding force push unless I'm absolutely sure.

Vanshitaaa20 committed:
fixed FunctionalCPD.py

ankurankan committed:
Merge branch 'dev' into fix/factors-doctests

ankurankan committed:
Merge branch 'fix/factors-doctests' of github.com:Vanshitaaa20/pgmpy into pr/2181

ankurankan committed:
format doctests in FactorSet.py

ankurankan committed:
Minor changes in factors/base.py

ankurankan committed:
Blackdoc on LinearGaussianCPD.py

ankurankan committed:
Remove sao2.cs file; Format noisyOr

ankurankan committed:
Fixes a few other files

ankurankan committed:
Revert some changes; and fix tests
